packages/rps/src/redux/game/saga.ts

```diff
@@ -39,7 +39,14 @@ const isPlayersTurnNext = (
 export function* gameSaga(client: RPSChannelClient) {
   const channel = yield actionChannel('*', buffers.fixed(10));
   while (true) {
-    yield take(channel);
+    const action = yield take(channel);
+
+    // TODO: @george @tom is this the right place for this?
+    if (action.type === 'Challenge') {
+      const {channelState} = yield select(getGameState);
+      yield challengeChannel(channelState, client);
+    }
```

Comment:
I think this is the right place, as all calls to the client live in this saga.

However, I am wondering if the call needs to provide the `channelState`, or even any data at all? If I am challenging, it is because it is not my turn: therefore my wallet should have all the information it needs (i.e. all the states) to launch the challenge because I have previously called `updateChannel` with them.

Reply:
To clarify: the API call only supplies the `channelId`, which seems correct.

But the `challengeChannel` generator function also puts an `'UpdateChannelState'` action, and I am wondering if we really need that?

Reply:
It actually only needs the `channelId` from that value